ts/src/htx.ts

```diff
@@ -5184,10 +5251,13 @@ export default class htx extends Exchange {
                 throw new ArgumentsRequired (this.id + ' createOrder() requires a triggerPrice for a stop order');
             }
         } else {
-            const defaultOperator = (side === 'sell') ? 'lte' : 'gte';
-            const stopOperator = this.safeString (params, 'operator', defaultOperator);
+            let triggerDirection = undefined;
+            [ triggerDirection, params ] = this.handleParamString (params, 'triggerDirection');
```

Comment:
isn't this a breaking change?

Reply:
the problem is that approach (setting `lte/gte` by `side`) was incorrect, so users relying on that incorrect approach "could be" breaking change. they should explicitly choose trigger direction and `side` is not related at all.

about `operator`, ok , i've added backward-compatibility for it.

Reply:
@ttodua Why was that considered incorret, if you place a trigger order, usually;

- if side is sell, you want sell when the price drops otherwise, you could place a limit order immediately in the OB if you tried to sell when the price was higher
- if side is buy, you want to buy when the price raises otherwise you could have the limit order in the oB waiting for the drop

Ofc exceptions might apply, but shouldn't this be the default behavior? (I don't have a definitive answer but I think both approaches have pros and cons just trying to consider them all)

Reply:
@ttodua No I'm not confusing with the stopLossPrice/takeProfitPrice orders. I'm referring to regular trigger orders, I'm not saying that trigger orders can't have both directions regardless of the side, I'm just asking if we should infer the direction by default to avoid adding a breaking change here and let the users provide `triggerDirection` if needed

Reply:
Exchanges like binance, automatically infer this direction by using the last price but we can't do that at ccxt because that's not available upon placing the order